contracts/pcv/utils/NamedStaticPCVDepositWrapper.sol

```diff
@@ -0,0 +1,180 @@
+
+    /// @notice the PCV balance
+    uint256 public override balance;
+
+    /// @notice the reported FEI balance
+    uint256 public feiReportBalance;
+
+    constructor(address _core, PCVDeposit[] memory newPCVDeposits) CoreRef(_core) {
+
+        // Uses oracle admin to share admin with CR oracle where this contract is used
+        _setContractAdminRole(keccak256("ORACLE_ADMIN_ROLE"));
+
+        // add all pcv deposits
+        for (uint256 i = 0; i < newPCVDeposits.length; i++) {
+            _addDeposit(newPCVDeposits[i]);
+        }
+    }
+
+    /// @notice helper method to delete a PCV deposit
+    function _removeDeposit(uint256 index) internal {
+        PCVDeposit storage removePCVDeposit = pcvDeposits[index];
+
+        uint256 depositBalance = removePCVDeposit.usdAmount;
+        uint256 feiDepositBalance = removePCVDeposit.feiAmount;
+
+        balance -= depositBalance;
+        feiReportBalance -= feiDepositBalance;
+        delete pcvDeposits[index];
+
+        emit BalanceUpdate(balance + depositBalance, balance, feiReportBalance + feiDepositBalance, feiReportBalance);
+        emit DepositRemoved(index);
+    }
+
+    /// @notice helper method to add a PCV deposit
+    function _addDeposit(PCVDeposit memory newPCVDeposit) internal {
+        require(newPCVDeposit.feiAmount > 0 || newPCVDeposit.usdAmount > 0, "NamedStaticPCVDepositWrapper: must supply either fei or usd amount");
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance += newPCVDeposit.usdAmount;
+        feiReportBalance += newPCVDeposit.feiAmount;
+        pcvDeposits.push(newPCVDeposit);
+
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    /// @notice helper method to edit a PCV deposit
+    function _editDeposit(
+        uint256 index,
+        uint256 usdAmount,
+        uint256 feiAmount,
+        uint256 underlyingTokenAmount,
+        string calldata depositName,
+        address underlying
+    ) internal {
+        PCVDeposit storage updatePCVDeposit = pcvDeposits[index];
+        int256 usdAmountDelta;
+        int256 feiAmountDelta;
+
+        unchecked {
+            // let this value go negative and not revert so that balance can go down if usdAmount is decreased
+            usdAmountDelta = usdAmount.toInt256() - updatePCVDeposit.usdAmount.toInt256();
+            feiAmountDelta = feiAmount.toInt256() - updatePCVDeposit.feiAmount.toInt256();
+        }
+
+        updatePCVDeposit.usdAmount = usdAmount;
+        updatePCVDeposit.depositName = depositName;
+        updatePCVDeposit.underlyingTokenAmount = underlyingTokenAmount;
+        updatePCVDeposit.underlying = underlying;
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance = (balance.toInt256() + usdAmountDelta).toUint256();
+        feiReportBalance = (feiReportBalance.toInt256() + feiAmountDelta).toUint256();
```

Comment:
subtract old then add new is a safer and more intuitive pattern than using unchecked integer arithmetic

Reply:
If we do that, it will be more gas inefficient as we will have to write to storage 2 times per deposit instead of the 1 time we currently have to.

Reply:
fixed